### ISSUE TITLE
Serious lag when using the cloud version for Hack The Box reports

### ISSUE THREAD

**Issue by vantascure** [Syslifters/sysreptor#370]
I faced some serious lag when using the cloud service to write a Hack The Box certificate report. It'd take a second or two for words to show after I'd typed them. I only faced the issue in the Internal Network Compromise section of the design. I believe it's due to the long detailed walkthrough I've written. My detailed walkthrough is 1993 lines and 13859 words. Switching to markdown only mode has made the lag more bearable, but not entirely resolved the issue. There's still some delay.

**Design:** HTB CPTS Report v1.1

**Comment by MWedl** [comment]
Hi,
this is a known problem in our markdown editor implementation. Long markdown texts take some time to parse and render to HTML. With increasing text length, the parsing time also increases. Unfortunately, markdown is hard to parse incrementally (because of some strange rules in the CommonMark specification), so we have to re-parse the whole document on changes resulting in lags for long documents. 

Some markdown editors implement markdown parsing while typing with (incremental) heuristics, but this approach leads to discrepancies between the editor highlighting and the rendered HTML. Because of this drawback, we decided against using (fast but inaccurate) heuristics.

**Comment by MWedl** [comment]
Additionally, to switching to markdown-only mode, you could also try to disable spell checks. This will not solve the root cause, but it might slightly increase responsiveness.

**Comment by aronmolnar** [comment]
We decided to split the mitigation into two parts.

1. Move the preview rendering process to JavaScript worker  
This is a performance improvement when using the split view. We will address this issue as one of our next work packs.
2. Introduce incremental parsing  
This addresses the root cause of the issue. It is, however, very complex to implement. Therefore, we cannot yet estimate when we will implement this.

**Comment by danymat** [comment]
Did you think on integrating other formats, such as .norg format (https://github.com/nvim-neorg/norg-specs/blob/main/1.0-specification.norg) ? That would not help in this specific case, but people could try specific formats if they want to have faster parsing and reliability.

**Comment by MWedl** [comment]
Implemented markdown preview rendering in a separate worker process in https://github.com/Syslifters/sysreptor/releases/tag/2025.12
